x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -244,7 +227,7 @@ public String toString() {
             + ", length="
             + tracker.getLength()
             + ", channel="
-            + (channel != null ? "yes" : "no")
+            + (channelRef != null ? "yes" : "no")
             + ", listeners="
             + listeners.size()
```

Comment:
I think that's fine, it's a `volatile` field so it's not going to be totally out of whack and the behavior is the same we had so far? Not great maybe but even if we sync on `listeners` I think we won't get a completely consistent view of things as we have no lock over all the fields used here (the `evicted` can still change async) so maybe it's not worth it? (my main contention with adding locking is that it would mess with debugging in potentially by introducing a new barrier depending on log settings)

Reply:
I believe `listeners` is no longer `volatile` with this PR? So could in theory be unboundedly wrong.

Also in theory, HashSet implementation could evolve in ways that make this fail in bad ways when concurrently modifying the listeners set (though hard to imagine it would ever do so).

Reply:
Urgh you're right I totally forgot that `listeners` isn't `volatile` any longer. I'll just `synchronize` this method on `listeners` for now as we do in all other spots where we touch `listeners`, it's just the only consistent way of doing things with the way it's implemented now IMO. It's probably not too necessary to do anything more complicated just for debugging purposes?